src/main/sensors/gyro.c

```diff
@@ -559,38 +556,111 @@ bool gyroInit(void)
+        lowpassFilter = gyroSensor->lowpass2Filter;
+        break;
+
+    default:
+        break;
+    }
+
+    // Establish some common constants
     const uint32_t gyroFrequencyNyquist = 1000000 / 2 / gyro.targetLooptime;
+    const float gyroDt = gyro.targetLooptime * 1e-6f;
+
+    // Gain could be calculated a little later as it is specific to the pt1/bqrcf2/fkf branches
+    const float gain = pt1FilterGain(lpfHz, gyroDt);
 
-    if (lpfHz && lpfHz <= gyroFrequencyNyquist) {  // Initialisation needs to happen once samplingrate is known
-        switch (gyroConfig()->gyro_soft_lpf_type) {
+    // If lowpass cutoff has been specified and is less than the Nyquist frequency
+    if (lpfHz && lpfHz <= gyroFrequencyNyquist) {
+        switch (type) {
+        case FILTER_PT1:
+            lowpassFilterApplyFn = (filterApplyFnPtr) pt1FilterApply;
+            for (int axis = 0; axis < XYZ_AXIS_COUNT; axis++) {
+                pt1FilterInit(&lowpassFilter[axis].pt1FilterState, gain);
+            }
+            break;
         case FILTER_BIQUAD:
-            gyroSensor->softLpfFilterApplyFn = (filterApplyFnPtr)biquadFilterApply;
+            lowpassFilterApplyFn = (filterApplyFnPtr) biquadFilterApply;
             for (int axis = 0; axis < XYZ_AXIS_COUNT; axis++) {
-                gyroSensor->softLpfFilterPtr[axis] = (filter_t *)&gyroSensor->softLpfFilter.gyroFilterLpfState[axis];
-                biquadFilterInitLPF(&gyroSensor->softLpfFilter.gyroFilterLpfState[axis], lpfHz, gyro.targetLooptime);
+                biquadFilterInitLPF(&lowpassFilter[axis].biquadFilterState, lpfHz, gyro.targetLooptime);
             }
             break;
-        case FILTER_PT1:
-            gyroSensor->softLpfFilterApplyFn = (filterApplyFnPtr)pt1FilterApply;
-            const float gyroDt = (float) gyro.targetLooptime * 0.000001f;
+        case FILTER_BIQUAD_RC_FIR2:
+            lowpassFilterApplyFn = (filterApplyFnPtr) biquadFilterApply;
             for (int axis = 0; axis < XYZ_AXIS_COUNT; axis++) {
-                gyroSensor->softLpfFilterPtr[axis] = (filter_t *)&gyroSensor->softLpfFilter.gyroFilterPt1State[axis];
-                pt1FilterInit(&gyroSensor->softLpfFilter.gyroFilterPt1State[axis], lpfHz, gyroDt);
+                biquadRCFIR2FilterInit(&lowpassFilter[axis].biquadFilterState, gain);
             }
             break;
-        default:
-#if defined(USE_FIR_FILTER_DENOISE)
-            // this should be case FILTER_FIR:
-            gyroSensor->softLpfFilterApplyFn = (filterApplyFnPtr)firFilterDenoiseUpdate;
+        case FILTER_CASCADED_BIQUADS:
+            lowpassFilterApplyFn = (filterApplyFnPtr) cascadedBiquadFilterApply;
+
+            int sections = 0;
+            if (order <= BIQUAD_LPF_ORDER_MAX) {
+                for (int axis = 0; axis < XYZ_AXIS_COUNT; axis++) {
+                    const int axisSections = biquadFilterLpfCascadeInit(
+                      &lowpassFilter[axis].biquadFilterSectionsState.biquad[order],
```

Comment:
pointer to first section in array: `lowpassFilter[axis].biquadFilterSectionsState.biquad`

Reply:
Handled in f639e5b